client/webserver/site/src/js/mmsettings.ts

```diff
@@ -2295,29 +2621,521 @@ class AssetPane {
     }
-    page.avail.textContent = Doc.formatFourSigFigs((dexAvail + cexAvail) / ui.conventional.conversionFactor)
-    if (assetID === feeAssetID) return
-    const { balance: { available: feeAvail } } = app().walletMap[feeAssetID]
-    page.feeAvail.textContent = Doc.formatFourSigFigs(feeAvail / feeUI.conventional.conversionFactor)
   }
+
+  for (const [key, value] of Object.entries(allocationResult.dex)) {
+    const assetID = Number(key)
+    if (assetID === baseID) setColor(page.dexBaseAlloc, value.status)
+    if (assetID === quoteID) setColor(page.dexQuoteAlloc, value.status)
+    if (assetID === baseFeeID) setColor(page.dexBaseFeeAlloc, value.status)
+    if (assetID === quoteFeeID) setColor(page.dexQuoteFeeAlloc, value.status)
+    dexBalances[assetID] = value.amount
+  }
+  for (const [key, value] of Object.entries(allocationResult.cex)) {
+    const assetID = Number(key)
+    if (assetID === baseID) setColor(page.cexBaseAlloc, value.status)
+    if (assetID === quoteID) setColor(page.cexQuoteAlloc, value.status)
+    cexBalances[assetID] = value.amount
+  }
+
+  const baseFeeNotTraded = baseFeeID !== baseID && baseFeeID !== quoteID
+  const quoteFeeNotTraded = quoteFeeID !== quoteID && quoteFeeID !== baseID
+
+  Doc.setVis(baseFeeNotTraded, page.baseFeeHeader, page.dexBaseFeeAlloc)
+  Doc.setVis(quoteFeeNotTraded && baseFeeID !== quoteFeeID, page.quoteFeeHeader, page.dexQuoteFeeAlloc)
+  Doc.setVis(cexName, page.cexRow)
+
+  const format = (v: number, unitInfo: UnitInfo) => v ? Doc.formatCoinValue(v, unitInfo) : '0'
+
+  page.dexBaseAlloc.textContent = format(dexBalances[baseID], baseUI)
+  page.dexQuoteAlloc.textContent = format(dexBalances[quoteID], quoteUI)
+  if (baseFeeNotTraded) page.dexBaseFeeAlloc.textContent = format(dexBalances[baseFeeID], baseFeeUI)
+  if (quoteFeeNotTraded) page.dexQuoteFeeAlloc.textContent = format(dexBalances[quoteFeeID], quoteFeeUI)
+
+  if (cexBalances && cexName) {
+    page.cexBaseAlloc.textContent = format(cexBalances[baseID], baseUI)
+    page.cexQuoteAlloc.textContent = format(cexBalances[quoteID], quoteUI)
+    setCexElements(div, cexName)
+  }
+
+  setMarketElements(div, baseID, quoteID, host)
+}
+
+type Fees = {
+  swap: number
+  redeem: number
+  refund: number
+  funding: number
+}
+
+interface PerLotBreakdown {
+  totalAmount: number
+  tradedAmount: number
+  fees: Fees
+  slippageBuffer: number
+  multiSplitBuffer: number
+}
+
+function newPerLotBreakdown () : PerLotBreakdown {
+  return {
+    totalAmount: 0,
+    tradedAmount: 0,
+    fees: { swap: 0, redeem: 0, refund: 0, funding: 0 },
+    slippageBuffer: 0,
+    multiSplitBuffer: 0
+  }
+}
+
+interface PerLot {
+  cex: Record<number, PerLotBreakdown>
+  dex: Record<number, PerLotBreakdown>
+}
+
+interface FeeReserveBreakdown {
+  buyReserves: Fees
+  sellReserves: Fees
+}
+
+type AllocationStatus = 'sufficient' | 'insufficient' | 'sufficient-with-rebalance'
+
+interface CalculationBreakdown {
```

Comment:
The `CalculationBreakdown` is currently unused, but it will be used in a follow-up to to show in the UI how the final allocation was calculated.